packages/blockchain-wallet-v4/src/redux/data/coinify/sagas.js

```diff
@@ -303,6 +304,27 @@ export default ({ api, options }) => {
     }
   }
 
+  const pollKYCsPending = function * () {
+    try {
+      const kycs = yield select(S.getKycs)
+      console.log(kycs)
+      const statusPath = x => path(['0', 'state'], x)
+      let status = kycs.map(kycs => statusPath(kycs)).getOrElse('pending')
+      while (equals(status, 'pending')) {
+        yield call(delay, 1000)
+        const coinify = yield call(getCoinify)
+        const kycs = yield apply(coinify, coinify.getKYCs)
+        const byTime = (a, b) => b.createdAt - a.createdAt
+        const sortedKYCs = sort(byTime, kycs)
+        status = statusPath(sortedKYCs)
+        yield put(A.getKYCsSuccess(sortedKYCs))
+      }
+    } catch (e) {
+      console.log('pollKYCsPending failure', e)
+      yield put(A.getKYCsFailure(e))
+    }
+  }
+
```

Comment:
Can we extract this out to a more generic polling function?

```
function poll(check, success) {
   if (!check) keepPolling()
   else success()
}```

Reply:
We could do that but I'm not sure it would be very beneficial given:
-a while loop wouldn't be much more complex than that function
-sometimes we might want to run the keepPolling function at least once (ie do while) and sometimes not
-it means we need to create another (generator) function for keepPolling (here we don't need any success function)
Also, we would need to pass the delay as an argument because we'll want different delays for different usecases of the poll function (if we ever need to use that function in other places

Reply:
Okay, we can revisit this in the future if we need multiple polling functions and they are all similar.